ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -139,6 +139,25 @@ struct Tensor {
 
     std::vector<Device*> get_workers(bool blocking = false) const;
 
+    // Converts a buffer of elements of type `T` to a `Tensor`.
+    // Elements are assumed to be stored in row-major order. The size of the span and the type have to match `spec`.
+    //
+    // TODO: tilized layouts and reduced precision types are currently not supported.
```

Comment:
I'd say it only supports RowMajor for bfloat16, float32, int32, right?
No support for
* Tile layout
* Sharding
* Other data types

Reply:
Also bfloat8/4/2_b is not supported in row major at all, conceptually

Reply:
can we mention what happens to data below?
* do we share data using borrowed storage? 
* or do we copy data and store into owned storage?
* if we do one of another - can we say there is always a copy except x,y cases?
* if we use borrowed storage, does a user have control to say "no", don't share my data?

Reply:
Added a comment - the data is currently copied into the tensor. But yes, we definitely need a way to create a borrowed buffer backed Tensor out of the buffer. And the clear API!